ast/ast.go

```diff
@@ -342,7 +342,7 @@ func Bool(tk *token.Token) *BoolNode {
 // Integer create node for integer value
 func Integer(tk *token.Token) *IntegerNode {
 	var v any
-	if num := token.ToNumber(tk.Value); num != nil {
+	if num, _ := token.ToNumber(tk.Value); num != nil {
```

Comment:
This looks odd

Shouldn't it be better and more logic now you refactored to have something like that

```suggestion
	if num, err := token.ToNumber(tk.Value); err == nil {
```

With code refactored to return an error when